velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -0,0 +1,110 @@
+      const std::unique_ptr<Reader> reader,
+      const RowTypePtr& rowType) {
+    auto rowReaderOpts = getReaderOpts(rowType);
+    auto scanSpec = makeScanSpec(rowType);
+    rowReaderOpts.setScanSpec(scanSpec);
+    auto rowReader = reader->createRowReader(rowReaderOpts);
+    return rowReader;
+  };
+
+  std::unique_ptr<facebook::velox::parquet::ParquetReader> createReaderInMemory(
+      const dwio::common::MemorySink& sink,
+      const dwio::common::ReaderOptions& opts) {
+    std::string_view data(sink.data(), sink.size());
+    return std::make_unique<facebook::velox::parquet::ParquetReader>(
+        std::make_unique<dwio::common::BufferedInput>(
+            std::make_shared<InMemoryReadFile>(data), opts.getMemoryPool()),
+        opts);
+  };
+};
+
+std::vector<CompressionKind> params = {
+    CompressionKind::CompressionKind_NONE,
+    CompressionKind::CompressionKind_SNAPPY,
+    CompressionKind::CompressionKind_ZSTD,
+    CompressionKind::CompressionKind_LZ4,
+    CompressionKind::CompressionKind_GZIP,
+};
+
+TEST_F(ParquetWriterTest, compression) {
+  auto schema =
+      ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6"},
+          {INTEGER(),
+           DOUBLE(),
+           BIGINT(),
+           INTEGER(),
+           BIGINT(),
+           INTEGER(),
+           DOUBLE()});
+  const int64_t kRows = 10'000;
+  const auto data = makeRowVector({
+      makeFlatVector<int32_t>(kRows, [](auto row) { return row + 5; }),
+      makeFlatVector<double>(kRows, [](auto row) { return row - 10; }),
+      makeFlatVector<int64_t>(kRows, [](auto row) { return row - 15; }),
+      makeFlatVector<uint32_t>(kRows, [](auto row) { return row + 20; }),
+      makeFlatVector<uint64_t>(kRows, [](auto row) { return row + 25; }),
+      makeFlatVector<int32_t>(kRows, [](auto row) { return row + 30; }),
+      makeFlatVector<double>(kRows, [](auto row) { return row - 25; }),
+  });
+
+  // Create an in-memory writer
+  auto sink = std::make_unique<MemorySink>(
+      200 * 1024 * 1024,
+      dwio::common::FileSink::Options{.pool = leafPool_.get()});
+  auto sinkPtr = sink.get();
+  facebook::velox::parquet::WriterOptions writerOptions;
+  writerOptions.memoryPool = leafPool_.get();
+
+  const auto& fieldNames = schema->names();
+
+  for (int i = 0; i < params.size(); i++) {
+    writerOptions.columnCompressionsMap[fieldNames[i]] = params[i];
+  }
+
+  auto writer = std::make_unique<facebook::velox::parquet::Writer>(
+      std::move(sink), writerOptions, rootPool_, schema);
+  writer->write(data);
+  writer->close();
+
+  dwio::common::ReaderOptions readerOptions{leafPool_.get()};
+  auto reader = createReaderInMemory(*sinkPtr, readerOptions);
+
+  ASSERT_EQ(reader->numberOfRows(), kRows);
+  ASSERT_EQ(*reader->rowType(), *schema);
+
+  for (int i = 0; i < params.size(); i++) {
+    EXPECT_EQ(
+        reader->getCompressionKind(0, i),
+        (i < params.size()) ? params[i] : common::CompressionKind_NONE);
```

Comment:
Can we use a non-default value for the compression?

Reply:
Done, using `CompressionKind::CompressionKind_SNAPPY` as the compression.